### PR TITLE
analytic events

### PR DESCRIPTION
**Pull request by xahgmah**
https://trello.com/c/DHE8iKRK/19-1-emits-analytics-events

### CODE REVIEW

File: video_xblock/static/js/videojs_event_plugin.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            .on('seeked', function () {
+                this.logEvent('onSeek')
+            });
+
+        this.log = function (eventName, data) {
+            data = data || {};
+            data['event_type'] = 'xblock-video.' + eventName;
+            parent.postMessage({
+                'action': 'analytics',
+                'event_data': data,
```
Comment (z4y4ts) [inline video_xblock/static/js/videojs_event_plugin.js]: use camelCase

File: video_xblock/static/js/videojs_event_plugin.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+             'onHideLanguageMenu',
+             'onShowTranscript',
+             'onHideTranscript',
+             'onShowCaptions',
+             'onHideCaptions',
+             */
+        ];
+
+        this.onReady = function () {
+            this.log('ready_video');
```
Comment (z4y4ts) [inline video_xblock/static/js/videojs_event_plugin.js]: This `_` and similar below concern me.
Are default player event titles use `_`, but not `-` or camelCase?

File: video_xblock/video_xblock.py [hunk 3]
```diff
@@ -233,6 +233,20 @@ def save_player_state(self, request, suffix=''):
...
         return {'success': True}
 
+    @XBlock.json_handler
+    def publish_event(self, data, suffix=''):
+        """
+        Handler to publish XBlock event from frontend.
+        Called by student_view's JavaScript
+        """
+        try:
+            event_type = data.pop('event_type')
```
Comment (z4y4ts) [inline video_xblock/video_xblock.py]: Why `.pop()`?

File: video_xblock/static/js/videojs_event_plugin.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        var currentTime = 0;
+
+        this.events = [
+            'onReady',
+            'onPlay',
+            'onPause',
+            'onEnded',
+            'onSpeedChange',
+             'onSeek',
+            /*TODO
```
Comment (z4y4ts) [inline video_xblock/static/js/videojs_event_plugin.js]: I understand why you want to comment these events. But the reason is not clear from the code.

If this plugin works with these lines uncommented don't comment these events.
If it doesn't work, add a note when to uncomment these events.

File: video_xblock/static/js/videojs_event_plugin.js (hunk 2 above)
Comment (xahgmah) [inline video_xblock/static/js/videojs_event_plugin.js]: Yes `_`
https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/js/src/video/09_events_plugin.js#L65

File: video_xblock/video_xblock.py (hunk 3 above)
Comment (xahgmah) [inline video_xblock/video_xblock.py]: CopyPaste from 
https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/drag_and_drop_v2.py#L447

**Pull request merged**